### PR TITLE
[SPARK-47375][Doc][FollowUp] Correct the preferTimestampNTZ option description in JDBC doc

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Correct the preferTimestampNTZ option description in JDBC doc as per https://github.com/apache/spark/pull/45496

### Why are the changes needed?

The current doc is wrong about the jdbc option preferTimestampNTZ

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Just doc change
### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @sadikovi

**Review by sadikovi — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master/3.5/3.4

File: docs/sql-data-sources-jdbc.md [hunk 1]
```diff
@@ -368,8 +368,9 @@ logging into the data sources.
     <td><code>preferTimestampNTZ</code></td>
     <td>false</td>
     <td>
-      When the option is set to <code>true</code>, all timestamps are inferred as TIMESTAMP WITHOUT TIME ZONE.
-      Otherwise, timestamps are read as TIMESTAMP with local time zone.
+      When the option is set to <code>true</code>, TIMESTAMP WITHOUT TIME ZONE type are inferred as Spark's TimestampNTZ type.
+      Otherwise, it is interpreted as Spark's Timestamp type(equivalent to TIMESTAMP WITHOUT LOCAL TIME ZONE).
```
Comment (cloud-fan) [inline docs/sql-data-sources-jdbc.md:372]: WITHOUT -> WITH

File: docs/sql-data-sources-jdbc.md [hunk 2]
```diff
@@ -368,8 +368,9 @@ logging into the data sources.
...
+      This setting specifically affects only the inference of TIMESTAMP WITHOUT TIME ZONE data type. Both TIMESTAMP WITHOUT LOCAL TIME ZONE and TIMESTAMP WITH TIME ZONE data types are consistently interpreted as Spark's Timestamp type regardless of this setting.
```
Comment (cloud-fan) [inline docs/sql-data-sources-jdbc.md:373]: ditto

File: docs/sql-data-sources-jdbc.md (hunk 1 above)
Comment (gengliangwang) [reply docs/sql-data-sources-jdbc.md:372]: Nice catch. Updated in https://github.com/apache/spark/pull/45510

**Comment by yaooqinn** [comment]
This seems incorrect to be marked as a follow-up to be backported branch-3.4/3.5

**Comment by gengliangwang** [comment]
> This seems incorrect to be marked as a follow-up to be backported branch-3.4/3.5

Yes, I noticed this during merging this PR...I should create a new Jira ticket at the beginning.

**Pull request closed without merging**